weston-ivi-shell/src/ivi-controller.c

```diff
@@ -1723,36 +1723,45 @@ surface_event_remove(struct wl_listener *listener, void *data)
     struct notification *not, *next;
 
```

Comment:
How about refactoring the code to two functions? (names below can be better):
-> remove_bkgnd_surface
-> remove_shell_surface

Implementation does look much cleaner then